### PR TITLE
Fixed slow pending-coinbase unit test

### PR DESCRIPTION
**Pull request by deepthiskumar**
Reduced the quickcheck trials, fixed the tree size [include the latency factor and take `ceil_log2` of the stacks required]

Explain how you tested your changes here.

Number of trials such that the test accesses (push a coinbase and delete a stack) all the stacks at least once.

Checklist:

- [ ] Tests were added for the new behavior
- [X] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [ ] Does this close issues?

### CODE REVIEW

**Comment by enolan** [comment]
I'll look at this when I get back from the doctor, I know some stuff about QuickCheck, may be able to improve things.

**Comment by deepthiskumar** [comment]
> I'll look at this when I get back from the doctor, I know some stuff about QuickCheck, may be able to improve things.

Just fyi, the tree size has reduced significantly so that should reduce the unit-test time as well.

**Comment by enolan** [comment]
`gen_imperative_list` from our `Quickcheck_lib` might be useful here

File: src/lib/coda_base/pending_coinbase.ml [hunk 1]
```diff
@@ -868,28 +866,30 @@ let%test_unit "push and pop multiple stacks" =
...
     in
     assert (Stack.equal popped_stack expected_stack) ;
     updated_pending_coinbases
   in
-  let add_remove_check trials =
-    fill trials ;
-    Queue.iter queue_of_stacks ~f:(fun expected_stack ->
-        pending_coinbases := check !pending_coinbases expected_stack ) ;
-    Queue.clear queue_of_stacks
+  let add_remove_check coinbases_list =
```
[the remark this answers is not in the document]
Comment (deepthiskumar) [reply src/lib/coda_base/pending_coinbase.ml]: That is a list of lists of coinbases. Each inner list is then used to create a pending coinbase stack. So maybe `coinbase_lists`?

**Pull request merged**